test/extended/security/service_serving_cert_signer.go

```diff
@@ -0,0 +1,165 @@
+package security
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	certutil "k8s.io/client-go/util/cert"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	serviceName               = "test-service"
+	secretName                = "test-secret"
+	configMapName             = "test-config"
+	servingCertAnnotation     = "service.alpha.openshift.io/serving-cert-secret-name"
+	configMapBundleAnnotation = "service.alpha.openshift.io/inject-cabundle"
+	configMapBundleDataKey    = "service-ca.crt"
+	signerCertNamespace       = "openshift-service-ca"
+	signerCertSecretName      = "service-serving-cert-signer-signing-key"
+)
+
+var _ = g.Describe("[Feature:ServiceServingCert] Services that request serving certificates", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc = exutil.NewCLI("service-serving-cert-signer", exutil.KubeConfigPath())
+		f  = oc.KubeFramework()
+	)
+
+	g.It("see a created secret and configmap", func() {
+		g.By("an annotated service resulting in a generated serving cert")
+		_, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Create(annotatedService())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer f.ClientSet.CoreV1().Services(f.Namespace.Name).Delete(serviceName, nil)
+
+		generatedSecret, err := pollForServiceServingSecret(f.ClientSet, f.Namespace.Name)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Delete(secretName, nil)
+
+		g.By("an annotated configmap injected with a signer CA bundle data item")
+		_, err = f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(annotatedConfigMap())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Delete(configMapName, nil)
+
+		injectedCABundle, err := pollForConfigMapCABundle(f.ClientSet, f.Namespace.Name)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Check that the injected CA bundle matches the signer CA
+		signer, err := f.ClientSet.CoreV1().Secrets(signerCertNamespace).Get(signerCertSecretName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		caCert, ok := signer.Data[kapiv1.TLSCertKey]
+		o.Expect(ok).NotTo(o.BeFalse())
+		o.Expect(caCert).To(o.BeEquivalentTo(injectedCABundle))
```

Comment:
once @sallyom's pr goes in, we should also test that the configmap has no other keys as well.